lib/Pipes/Lift.cpp

```diff
@@ -0,0 +1,158 @@
+  LibHelpersPath = OptionalHelpers.value();
+
+  std::string EarlyLinkedName = "/share/revng/early-linked-" + ArchName + ".ll";
+  auto OptionalEarlyLinked = ResourceFinder.findFile(EarlyLinkedName);
+  revng_assert(OptionalEarlyLinked.has_value(), "Cannot find early-linked.ll");
+  EarlyLinkedPath = OptionalEarlyLinked.value();
+}
+
+/// Given an architecture name, loads the appropriate version of the PTC
+/// library, and initializes the PTC interface.
+///
+/// \param Architecture the name of the architecture, e.g. "arm".
+/// \param PTCLibrary a reference to the library handler.
+///
+/// \return EXIT_SUCCESS if the library has been successfully loaded.
+static int loadPTCLibrary(LibraryPointer &PTCLibrary) {
+  ptc_load_ptr_t ptc_load = nullptr;
+  void *LibraryHandle = nullptr;
+
+  // Look for the library in the system's paths
+  LibraryHandle = dlopen(LibTinycodePath.c_str(), RTLD_LAZY | RTLD_NODELETE);
+
+  if (LibraryHandle == nullptr) {
+    fprintf(stderr, "Couldn't load the PTC library: %s\n", dlerror());
+    return EXIT_FAILURE;
+  }
+
+  // The library has been loaded, initialize the pointer, the caller will take
+  // care of dlclose it from now on
+  PTCLibrary.reset(LibraryHandle);
+
+  // Obtain the address of the ptc_load entry point
+  ptc_load = reinterpret_cast<ptc_load_ptr_t>(dlsym(LibraryHandle, "ptc_load"));
+
+  if (ptc_load == nullptr) {
+    fprintf(stderr, "Couldn't find ptc_load: %s\n", dlerror());
+    return EXIT_FAILURE;
+  }
+
+  // Initialize the ptc interface
+  if (ptc_load(LibraryHandle, &ptc) != 0) {
+    fprintf(stderr, "Couldn't find PTC functions.\n");
+    return EXIT_FAILURE;
+  }
+
+  return EXIT_SUCCESS;
+}
+
+void pipeline::LiftPipe::run(Context &Ctx,
+                             const TemporaryFile &SourceBinary,
+                             LLVMContainer &TargetsList) {
+  if (not SourceBinary.exists())
+    return;
+
+  auto &Model = getWritableModelFromContext(Ctx);
+  revng_check(BaseAddress % 4096 == 0, "Base address is not page aligned");
+
+  BinaryFile TheBinary(SourceBinary.path()->str(), BaseAddress);
+
+  findFiles(TheBinary.architecture().name());
+
+  // Load the appropriate libtyncode version
+  LibraryPointer PTCLibrary;
+  auto Result = loadPTCLibrary(PTCLibrary);
+  revng_assert(Result == EXIT_SUCCESS);
+
+  // Translate everything
+  Architecture TargetArchitecture;
+  CodeGenerator Generator(TheBinary,
+                          TargetArchitecture,
+                          &TargetsList.getModule(),
+                          Model,
+                          LibHelpersPath,
+                          EarlyLinkedPath);
+
+  llvm::Optional<uint64_t> EntryPointAddressOptional;
+  if (EntryPointAddress.getNumOccurrences() != 0)
+    EntryPointAddressOptional = EntryPointAddress;
+  Generator.translate(EntryPointAddressOptional);
+}
+
+using namespace pipeline;
+
+static RegisterPipe<LiftPipe> E1("Lift");
```

Comment:
I suggest that we:
1. Rename `LiftPipe::Name` to `::Description`
2. Move the registered name to `LiftPipe::Name`
3. Use lower-case [Kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) for registered names, as LLVM does for passes.

Reply:
I see, but it'd be so much better to have a the pipe name in the class. Otherwise one has to dig into the `.cpp` files. It's what happens with LLVM too, but at least there we have `--help` to list all passes.

What if we informally move it in the class and 
```suggestion
static RegisterPipe<LiftPipe> E1(LiftPipe::Name);
```

Those that need to be registered multiples times simply don't have `::Name`.

Reply:
what i did was to add a overload to the various register things that does not take the string and deduces it from the registerd type. So one can overload it if he wishes